docs/development-environment-setup.md

```diff
@@ -0,0 +1,99 @@
+# Development Environment Setup
+
+Before you get started with contributing to LocalStack, make sure you’ve familiarized yourself with LocalStack from the perspective of a user.
+You can follow our [getting started guide](https://docs.localstack.cloud/get-started/).
+Once LocalStack runs in your Docker environment and you’ve played around with the LocalStack and `awslocal` CLI, you can move forward to set up your developer environment.
+
+## Development requirements
+
+You will need the following tools for the local development of LocalStack.
+
+* [Python 3.11+](https://www.python.org/downloads/) and `pip`
+* [Node.js & npm](https://nodejs.org/en/download/)
+* [Docker](https://docs.docker.com/desktop/)
+
+We recommend you to individually install the above tools using your favorite package manager.
+For example, on macOS, you can use [Homebrew](https://brew.sh/) to install the above tools.
+
+### Setting up the Development Environment
+
+To make contributions to LocalStack, you need to be able to run LocalStack in host mode from your IDE, and be able to attach a debugger to the running LocalStack instance.
+We have a basic tutorial to cover how you can do that.
+
+The basic steps include:
+
+1. Fork the localstack repository on GitHub [https://github.com/localstack/localstack/](https://github.com/localstack/localstack/)
+2. Clone the forked localstack repository `git clone git@github.com:<GITHUB_USERNAME>/localstack.git`
+3. Ensure you have `python`, `pip`, `node`, and `npm` installed.
+   > You might also need `java` for some emulated services.
+4. Install the Python dependencies using `make install`.
+   > This will install the required pip dependencies in a local Python 3 `venv` directory called `.venv` (your global python packages will remain untouched).
+   > Depending on your system, some `pip` modules may require additional native libs installed. 
+5. Start localstack in host mode using `make start`
+
+{{< youtube XHLBy6VKuCM >}}
```

Comment:
Here is a Hugo shortcode, and there are at least 13 others like this (looking for `{{<`).